src/orion/core/utils/random_state.py

```diff
@@ -77,3 +78,22 @@ def seed(cls, base_seed: int | None) -> RandomState:
             # Add the base_seed property in this case.
             random_state = replace(random_state, base_seed=base_seed)
         return random_state
+
+
+@contextmanager
+def control_randomness(random_state: RandomState | None):
+    """Seeds the randomness inside the indented block of code using `self.random_state`."""
+    if random_state is None:
+        yield
+        return
+
+    # Save the initial random state.
+    initial_rng_state = RandomState.current()
+    # Set the random state.
+    random_state.set()
+    yield
+    # Update the random state stored on `self`, so that the changes inside the block are
+    # reflected in the RandomState object.
+    random_state = RandomState.current()
```

Comment:
This line isn't used, now that I look at it.

Reply:
Oh, you're right! Does it really make sense to have the RandomState be an immutable object? If we want to track the RNG states that's by definition tracking mutable objects.

Reply:
Fixed in d3e4935e